src/models/stores/sortable-documents.ts

```diff
@@ -0,0 +1,196 @@
+import { ObservableSet, makeAutoObservable, runInAction } from "mobx";
+import { DocumentModelType } from "../document/document";
+import { isSortableType } from "../document/document-types";
+import { IBaseStores } from "./base-stores-types";
+import { DocumentsModelType } from "./documents";
+import { GroupsModelType } from "./groups";
+import { ClassModelType } from "./class";
+import { DB } from "../../lib/db";
+
+
+type SortedDocument = {
+  sectionLabel: string;
+  documents: DocumentModelType[];
+}
+
+type TagWithDocs = {
+  tagKey: string;
+  tagValue: string;
+  docKeysFoundWithTag: string[];
+};
+
+export class SortedDocuments {
+  stores: IBaseStores;
```

Comment:
Rather than using `IBaseStores` it is better to define an interface that just includes the properties from IBaseStores that this class is using. This cuts out a possible circular dependency, and it also makes it easier to test this store in isolation. The test would only need to pass in mock versions of the stores that this class is using instead of having to mock all of the properties of `IBaseStores`.